docs/advanced.rst

```diff
@@ -390,57 +390,48 @@ It can be used to specify multiple categories also.
 Pipenv includes the `safety <https://github.com/pyupio/safety>`_ package, and will use it to scan your dependency graph
 for known security vulnerabilities!
 
-Example::
-
-    $ cat Pipfile
-    [packages]
-    django = "==1.10.1"
-
-    $ pipenv check
-    Checking PEP 508 requirements...
-    Passed!
-    Checking installed package safety...
-
-    33075: django >=1.10,<1.10.3 resolved (1.10.1 installed)!
-    Django before 1.8.x before 1.8.16, 1.9.x before 1.9.11, and 1.10.x before 1.10.3, when settings.DEBUG is True, allow remote attackers to conduct DNS rebinding attacks by leveraging failure to validate the HTTP Host header against settings.ALLOWED_HOSTS.
-
-    33076: django >=1.10,<1.10.3 resolved (1.10.1 installed)!
-    Django 1.8.x before 1.8.16, 1.9.x before 1.9.11, and 1.10.x before 1.10.3 use a hardcoded password for a temporary database user created when running tests with an Oracle database, which makes it easier for remote attackers to obtain access to the database server by leveraging failure to manually specify a password in the database settings TEST dictionary.
+By default ``pipenv check`` will scan the virtualenv packages for what is installed and use this as the input to the safety command.
+To have ``pipenv check`` use the ``Pipfile.lock`` to determine package inputs simply pass ``--use-lock`` to use the default packages
+category, or pass the specific ``--categories`` you want to check against.
```

Comment:
IMO This new behaviour should be the default.
scanning of installed packages, means to me,  everything we install in a project.
Everything includes implicit packages which found in the lock file, and explicit packages we
specify in the Pipfile.

Reply:
Well the lock file covers also what it is in the `Pipfile`, but my concern with making this the default behavior is any existing workflows that may depend on scanning what is installed, since its totally possible and reasonable to sometimes install an extra package outside the view of Pipfile/Pipfile.lock   so I was reluctant to change the default behavior here.   Plus it would only scan the default packages and none of the other named categories without supplying the `--categories` argument.  

I do plan to support the ability to specify an env var in the future that would set the `--categories` implicitly, but I haven't gotten there yet, if specified it would complicate this further since there wouldn't be a way to fall back to what is actually installed.  Let me think on this some more.